js/src/modals/SMSVerification/SMSVerification.js

```diff
@@ -0,0 +1,166 @@
+
+import {
+  GATHERING_DATA, GATHERED_DATA,
+  POSTING_REQUEST, POSTED_REQUEST,
+  REQUESTING_SMS, REQUESTED_SMS,
+  POSTING_CONFIRMATION, POSTED_CONFIRMATION,
+  DONE
+} from './store';
+
+import GatherData from './GatherData';
+import SendRequest from './SendRequest';
+import QueryCode from './QueryCode';
+import SendConfirmation from './SendConfirmation';
+import Done from './Done';
+
+@observer
+export default class SMSVerification extends Component {
+  static propTypes = {
+    store: PropTypes.any.isRequired,
+    account: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired
+  }
+
+  static phases = { // mapping (store steps -> steps)
+    [GATHERING_DATA]: 0, [GATHERED_DATA]: 0,
+    [POSTING_REQUEST]: 1, [POSTED_REQUEST]: 1, [REQUESTING_SMS]: 1,
+    [REQUESTED_SMS]: 2,
+    [POSTING_CONFIRMATION]: 3, [POSTED_CONFIRMATION]: 3,
+    [DONE]: 4
+  }
+
+  componentDidMount () {
+    this.props.store.gatherData();
+  }
+
+  render () {
+    const phase = SMSVerification.phases[this.props.store.step];
+    const { error, isStepValid } = this.props.store;
+
+    return (
+      <Modal
+        actions={ this.renderDialogActions(phase, error, isStepValid) }
+        title='verify your account via SMS'
+        visible scroll
+        current={ phase }
+        steps={ ['Enter Data', 'Request', 'Enter Code', 'Confirm', 'Done!'] }
+        waiting={ error ? [] : [ 1, 3 ] }
+      >
+        { this.renderStep(phase, error) }
+      </Modal>
+    );
+  }
+
+  renderDialogActions (phase, error, isStepValid) {
+    const { store, account, onClose } = this.props;
+
+    const cancel = (
+      <Button
+        key='cancel' label='Cancel'
+        icon={ <ContentClear /> }
+        onClick={ onClose }
+      />
+    );
+    if (error) return (<div>{ cancel }</div>);
+
+    if (phase === 4) {
+      return (
+        <div>
+          { cancel }
+          <Button
+            key='done' label='Done'
+            disabled={ !isStepValid }
+            icon={ <ActionDoneAll /> }
+            onClick={ onClose }
+          />
+        </div>
+      );
+    }
+
+    let action;
+    if (phase === 3) {
+      action = store.done;
+    } else if (phase === 2) {
+      action = store.sendConfirmation;
+    } else if (phase === 1) {
+      action = store.queryCode;
+    } else if (phase === 0) {
+      action = store.sendRequest;
+    }
+
+    return (
+      <div>
+        { cancel }
+        <Button
+          key='next' label='Next'
+          disabled={ !isStepValid }
+          icon={ <IdentityIcon address={ account } button /> }
+          onClick={ action }
+        />
+      </div>
+    );
+  }
+
+  renderStep (phase, error) {
+    if (error) return (<p>{ error }</p>);
+
+    const {
+      step,
+      fee, number, isNumberValid, isVerified, hasRequested,
+      requestTx, isCodeValid, confirmationTx,
+      setNumber, setConsentGiven, setCode
+    } = this.props.store;
+
+    if (phase === 4) {
```

Comment:
Any reason why we are not using a `switch` or `else if` here? (As it stands, single blocks should probably be on seperate lines, unless it follows the chain with an `else if`)